test/integration/BCowPool.t.sol

```diff
@@ -0,0 +1,181 @@
+
+    // deal controller
+    deal(address(dai), controller, HUNDRED_UNITS);
+    deal(address(weth), controller, HUNDRED_UNITS);
+
+    // deal swapper
+    deal(address(weth), swapper.addr, HUNDRED_UNITS);
+
+    vm.startPrank(controller);
+    // deploy
+    // TODO: deploy with BCoWFactory
+    pool = new BCoWPool(address(settlement));
+    // bind
+    dai.approve(address(pool), type(uint256).max);
+    weth.approve(address(pool), type(uint256).max);
+    IBPool(pool).bind(address(dai), HUNDRED_UNITS, 2e18);
+    IBPool(pool).bind(address(weth), HUNDRED_UNITS, 8e18);
+    // finalize
+    IBPool(pool).finalize();
+    // enable trading
+    pool.enableTrading(APP_DATA);
+  }
+
+  function testBCowPoolSwap() public {
+    uint256 buyAmount = HUNDRED_UNITS;
+    uint256 sellAmount = calcOutGivenIn({
+      tokenBalanceIn: dai.balanceOf(address(pool)),
+      tokenWeightIn: pool.getDenormalizedWeight(address(dai)),
+      tokenBalanceOut: weth.balanceOf(address(pool)),
+      tokenWeightOut: pool.getDenormalizedWeight(address(weth)),
+      tokenAmountIn: buyAmount,
+      swapFee: 0
+    });
+
+    uint32 latestValidTimestamp = uint32(block.timestamp) + MAX_ORDER_DURATION - 1;
+
+    // swapper approves weth to vaultRelayer
+    vm.startPrank(swapper.addr);
+    weth.approve(settlement.vaultRelayer(), type(uint256).max);
+
+    // swapper creates the order
+    GPv2Order.Data memory swapperOrder = GPv2Order.Data({
+      sellToken: weth,
+      buyToken: dai,
+      receiver: GPv2Order.RECEIVER_SAME_AS_OWNER,
+      sellAmount: sellAmount, // WETH
+      buyAmount: buyAmount, // 100 DAI
+      validTo: latestValidTimestamp,
+      appData: APP_DATA,
+      feeAmount: 0,
+      kind: GPv2Order.KIND_BUY,
+      partiallyFillable: false,
+      buyTokenBalance: GPv2Order.BALANCE_ERC20,
+      sellTokenBalance: GPv2Order.BALANCE_ERC20
+    });
+
+    // swapper signs the order
+    (uint8 v, bytes32 r, bytes32 s) =
+      vm.sign(swapper.privateKey, GPv2Order.hash(swapperOrder, settlement.domainSeparator()));
+    bytes memory swapperSig = abi.encodePacked(r, s, v);
+
+    // order for bPool is generated
+    GPv2Order.Data memory poolOrder = GPv2Order.Data({
+      sellToken: dai,
+      buyToken: weth,
+      receiver: GPv2Order.RECEIVER_SAME_AS_OWNER,
+      sellAmount: buyAmount, // 100 DAI
+      buyAmount: sellAmount, // WETH
+      validTo: latestValidTimestamp,
+      appData: APP_DATA,
+      feeAmount: 0,
+      kind: GPv2Order.KIND_SELL,
+      partiallyFillable: true,
+      sellTokenBalance: GPv2Order.BALANCE_ERC20,
+      buyTokenBalance: GPv2Order.BALANCE_ERC20
+    });
+    bytes memory poolSig = abi.encode(poolOrder);
+
+    // solver prepares for call settle()
+    IERC20[] memory tokens = new IERC20[](2);
+    tokens[0] = IERC20(weth);
+    tokens[1] = IERC20(dai);
+
+    uint256[] memory clearingPrices = new uint256[](2);
+    // TODO: we can use more accurate clearing prices here
```

Comment:
is ok, i think ideally we could merge this test with `PoolSwap.t.sol`, in a "regression" way, such that we have the same expectations for swaps, but let's make it on another PR (and perhaps after BCoWFactory) is done